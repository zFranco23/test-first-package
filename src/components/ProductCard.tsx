import React from 'react';
import { createContext } from 'react';

import { useProduct } from '../hooks/useProduct';
import { ProductContextProps, Product, onChangeArgs, InitialValues} from '../interfaces/interfaces';

import styles from '../styles/styles.module.css';

export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;



export interface Props {
    product: Product;
    // children?: React.ReactElement | React.ReactElement[];
    children: (maxCount?: number) => JSX.Element;
    className?: string;
    style?: React.CSSProperties;
    onChange?: ( args: onChangeArgs ) => void;
    value?: number;
    initialValues?: InitialValues
}


export const ProductCard = (props: Props ) => {
    const { children, product, className, style, onChange, value, initialValues } = props;

    const { counter, increaseBy, maxCount } = useProduct({ onChange, product, value, initialValues });

    return (
        <Provider value={{
            counter,
            increaseBy,
            product,
            maxCount
        }}>
            <div 
                className={ `${ styles.productCard } ${ className }` }
                style={ style }
            >
                { children(initialValues?.maxCount) }
            </div>
        </Provider>
    )
}
