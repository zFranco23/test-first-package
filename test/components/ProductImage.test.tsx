import renderer from 'react-test-renderer';
import React from 'react';

import { product2 } from '../mock/products';
import { ProductCard } from '../../src/components/ProductCard';
import { ProductImage } from '../../src/components/ProductImage';

describe('ProductImage', () => {
  test('Should show correctly the image', () => {
    const wrapper = renderer.create(<ProductImage img="/abcd.jpg" />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Should show correctly the image', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
