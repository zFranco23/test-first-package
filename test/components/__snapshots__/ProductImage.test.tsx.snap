// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Should show correctly the image 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="/abcd.jpg"
/>
`;

exports[`ProductImage Should show correctly the image 2`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
